### DISCUSSION TITLE
Unable to log in after upgrade, even after reset of password from command line

### DISCUSSION THREAD

**Discussion by jpeterse** [inventree/InvenTree#8353] [General]
I haven't used my InvenTree for a while, and was going to start using it again.
Before I did, I did an upgrade of the server.

It was a big jump, from before the refactor into backend/frontend folder structure.
So I installed a fresh bare metal install in a new location, and I have the server running again, and successfully ran the update of the database as well.

But I can't log in. I get invalid username/password on the login page.
I have reset the password from command line. I have also invoked superuser command to add a new super user, and I can see the accounts in the database. But I still can't log in.

I have set DEBUG: true on config, and also set log options to DEBUG. But I can't find where it creates any log files.
So step 1: how can I get it to generate debug logs?

**Comment by ab1g17** [comment]
docker logs "name of container" (invetree-server/inventree-db etc) will show you the debug logs

**Reply by jpeterse** [reply]
Thanks. but it's a bare metal install. It's not in docker

**Comment by SchrodingersGat** [comment]
> Before I did, I did an upgrade of the server.
> It was a big jump, from before the refactor into backend/frontend folder structure.

As a bare minimum starting point you should review the release notes for all the intervening releases, and look at "breaking changes". There have been a number of updates which require you to make some changes to your configuration.

**Reply by jpeterse** [reply]
One of the reasons I did a clean install in a different location. Only the database was upgraded. The upgrade script didn't show any errors, so I think the DB updated ok.

**Comment by jpeterse** [comment]
getting closer. I decided to start again. Remove the apt package, and reinstalled.
This time, I created a new database, and got a blank instance of Inventree running.
Then I switched the database back to the old database, and I can log in now, and see all the data.

In the process I did run into two oddities.
1. When running invoke update, doesn't use the configuration from the deb package, but creates a new from template. I copied the config file from /etc to resolve this. But it does leave me wondering, which config file that it is using normally.
2. Second time I ran invoke worker I got an key signing issue:
`Failed to push task to queue
Traceback (most recent call last):
  File "/opt/inventree/env/lib/python3.9/site-packages/django_q/pusher.py", line 56, in pusher
    task = SignedPackage.loads(task[1])
  File "/opt/inventree/env/lib/python3.9/site-packages/django_q/signing.py", line 25, in loads
    return signing.loads(
  File "/opt/inventree/env/lib/python3.9/site-packages/django_q/core_signing.py", line 44, in loads
    TimestampSigner(key=key, salt=salt).unsign(s, max_age=max_age)
  File "/opt/inventree/env/lib/python3.9/site-packages/django_q/core_signing.py", line 80, in unsign
    result = super(TimestampSigner, self).unsign(value)
  File "/opt/inventree/env/lib/python3.9/site-packages/django_q/core_signing.py", line 65, in unsign
    raise BadSignature('Signature "%s" does not match' % sig)
django.core.signing.BadSignature: Signature "0DsBQrOGgcLD5KszKyMTTBm-AdzNRqWwBIdVvqlYx_0" does not match`
I have not seen the error after this

Last thing remaining to resolve, when I start Inventree as a service, it can't read the static files. It's reads them just fine with launched with invoke.

**Comment by jpeterse** [comment]
Server is up and running again.
Two additional findings this time around:
1. using default NGINX configuration file, NGINX was trying to forward to IPv6 address. Quick fix was to set the forward url to IPv4 address instead of localhost hostname
2. I noticed the default config.yaml has an entry "site-url: http://localhost:8000". But I think the default port is 6000 (as set in NGINX conf)?

Even though the server is running, and I have debug turned on, it is still not producing any logs. Logs are created during installation, but not when Inventree is running as a service. This one, I really don't know how to address.
